### PR TITLE
Feat/release 2.1.0

### PR DESCRIPTION
**Pull request by repjarms**
### Summary

This PR introduces updates for the `2.1.0` version of the Go SDK, specifically:

#### NLU

New CRUD methods for SentimentModels, ClassificationModels, and CategoriesModels

#### Text to speech

listCustomPrompts method added
addCustomPrompts method added
getCustomPrompts method added
deleteCustomPrompts method added
listSpeakerModels method added
createSpeakerModel method added
getSpeakerModel method added
deleteSpeakerModel method added

### CODE REVIEW

File: v2/assistantv1/assistant_v1.go [hunk 1]
```diff
@@ -253,11 +256,13 @@ func (assistant *AssistantV1) MessageWithContext(ctx context.Context, messageOpt
 	if err != nil {
 		return
 	}
-	err = core.UnmarshalModel(rawResponse, "", &result, UnmarshalMessageResponse)
```
Comment (apaparazzi0329) [inline v2/assistantv1/assistant_v1.go:256]: What is the reason fo these strange changes?

File: v2/assistantv1/assistant_v1.go [hunk 2]
```diff
@@ -3633,10 +3717,10 @@ func UnmarshalAgentAvailabilityMessage(m map[string]json.RawMessage, result inte
 // BulkClassifyOptions : The BulkClassify options.
 type BulkClassifyOptions struct {
 	// Unique identifier of the workspace.
-	WorkspaceID *string `json:"workspace_id" validate:"required,ne="`
```
Comment (apaparazzi0329) [inline v2/assistantv1/assistant_v1.go:3636]: What is going on here?

File: v2/assistantv1/assistant_v1.go (hunk 1 above)
Comment (repjarms) [reply v2/assistantv1/assistant_v1.go:256]: I wouldn't call the changes strange - it adds another level of `nil` checking when unmarhsalling to ensure that we only attempt an unmarshal if `rawResponse` is a non-nil value.

File: v2/assistantv1/assistant_v1.go (hunk 2 above)
Comment (repjarms) [reply v2/assistantv1/assistant_v1.go:3636]: It's not removed, you can see the diff on line `3720`. For all options models parameters, the new generate code removes the `json` tags, which are used in Go to define how to marshal/unmarshal properties to/from JSON.

Within each function definition, you can see how the properties within the options model are handled. Specifically, we `nil` check each property and then based on the API definition it is appended to structs that become query parameters, form-data, or body parameters.

File: v2/comparecomplyv1/compare_comply_v1.go [hunk 3]
```diff
@@ -2181,34 +2206,34 @@ func (options *ConvertToHTMLOptions) SetHeaders(param map[string]string) *Conver
 // CreateBatchOptions : The CreateBatch options.
 type CreateBatchOptions struct {
 	// The Compare and Comply method to run across the submitted input documents.
-	Function *string `json:"function" validate:"required"`
+	Function *string `validate:"required"`
 
 	// A JSON file containing the input Cloud Object Storage credentials. At a minimum, the credentials must enable `READ`
 	// permissions on the bucket defined by the `input_bucket_name` parameter.
-	InputCredentialsFile io.ReadCloser `json:"input_credentials_file" validate:"required"`
+	InputCredentialsFile io.ReadCloser `validate:"required"`
```
Comment (mediumTaj) [inline v2/comparecomplyv1/compare_comply_v1.go:2213]: just want to confirm that we dont need param name here anymore

File: v2/discoveryv2/discovery_v2_integration_test.go [hunk 4]
```diff
@@ -311,3 +311,17 @@ func TestEnrichments(t *testing.T) {
 
 	assert.Nil(t, deleteErr)
 }
+
+func TestQueryCollectionNotices(t *testing.T) {
```
Comment (mediumTaj) [inline v2/discoveryv2/discovery_v2_integration_test.go:315]: Do we have full disco v2 tests? are we testing with beta instance?

**Review by mediumTaj — APPROVED** [review]
👍 looks good! couple of questions but im sure its fine

File: v2/assistantv1/assistant_v1_suite_test.go [hunk 5]
```diff
@@ -17,9 +17,10 @@
 package assistantv1_test
 
 import (
+	"testing"
+
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	"testing"
 )
```
Comment (kevinkowa) [inline v2/assistantv1/assistant_v1_suite_test.go:24]: is the only change on this file just adding a new line?

File: v2/assistantv2/assistant_v2_suite_test.go [hunk 6]
```diff
@@ -17,9 +17,10 @@
 package assistantv2_test
 
 import (
+	"testing"
+
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	"testing"
 )
```
Comment (kevinkowa) [inline v2/assistantv2/assistant_v2_suite_test.go:24]: same here

File: v2/comparecomplyv1/compare_comply_v1.go (hunk 3 above)
Comment (repjarms) [reply v2/comparecomplyv1/compare_comply_v1.go:2213]: @mediumTaj no - there is only one case where we need to keep the JSON tags and that is for the recognition options model in STT due to Websockets.

File: v2/assistantv1/assistant_v1_suite_test.go (hunk 5 above)
Comment (kevinkowa) [reply v2/assistantv1/assistant_v1_suite_test.go:24]: I see this on multiple suite_test files, so I'm just going to mention it on this comment

File: v2/discoveryv2/discovery_v2_integration_test.go (hunk 4 above)
Comment (repjarms) [reply v2/discoveryv2/discovery_v2_integration_test.go:315]: We have a full test suite of generated unit tests, and then select integration tests. We are testing with beta.

File: v2/visualrecognitionv4/visual_recognition_v4.go [hunk 7]
```diff
@@ -1811,6 +1842,12 @@ func (options *CreateCollectionOptions) SetDescription(description string) *Crea
 	return options
 }
 
+// SetTrainingStatus : Allow user to set TrainingStatus
+func (options *CreateCollectionOptions) SetTrainingStatus(trainingStatus *TrainingStatus) *CreateCollectionOptions {
+	options.TrainingStatus = trainingStatus
+	return options
+}
+
```
Comment (kevinkowa) [inline v2/visualrecognitionv4/visual_recognition_v4.go:1850]: are we allowing the user to set training status here?

File: v2/assistantv1/assistant_v1_suite_test.go (hunk 5 above)
Comment (repjarms) [reply v2/assistantv1/assistant_v1_suite_test.go:24]: Yes the linter decided that standard library imports should be above package imports

File: v2/visualrecognitionv4/visual_recognition_v4.go (hunk 7 above)
Comment (repjarms) [reply v2/visualrecognitionv4/visual_recognition_v4.go:1850]: @kevinkowa good catch - I think the hand edit that removed this got edited out. I'll remove this

Comment (repjarms) [reply v2/visualrecognitionv4/visual_recognition_v4.go]: updated

**Review by kevinkowa — APPROVED** [review]
LGTM

File: v2/speechtotextv1/speech_to_text_v1_integration_test.go [hunk 8]
```diff
@@ -404,11 +405,17 @@ func (cb myCallBack) OnClose() {
 func (cb myCallBack) OnData(resp *core.DetailedResponse) {
 	var speechResults speechtotextv1.SpeechRecognitionResults
 	result := resp.GetResult().([]byte)
+
+	fmt.Println(resp)
```
Comment (apaparazzi0329) [inline v2/speechtotextv1/speech_to_text_v1_integration_test.go:409]: Do we want these print statements here? As well as the fmt import?

**Review by apaparazzi0329 — APPROVED** [review]
LGTM! Only one comment to be addressed

**Review by padamstx — APPROVED** [review]
LGTM

File: v2/speechtotextv1/speech_to_text_v1_integration_test.go (hunk 8 above)
Comment (repjarms) [reply v2/speechtotextv1/speech_to_text_v1_integration_test.go:409]: Nice catch I will remove

Comment (repjarms) [reply v2/speechtotextv1/speech_to_text_v1_integration_test.go]: removed

**Pull request merged**